clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1603,6 +1604,21 @@ void PointerVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV) {
   }
 }
 
+Atom *PointerVariableConstraint::getAtom(unsigned AtomIdx, Constraints &CS) {
+  if (AtomIdx < vars.size()) {
+    // If index is in bounds, just return the atom.
+    return vars[AtomIdx];
+  } else if (IsGeneric && AtomIdx == vars.size()) {
+    // Polymorphic types don't know how "deep" their pointers are beforehand so,
+    // we need to create new atoms for new pointer levels on the fly.
+    std::string Stars(vars.size(), '*');
+    Atom *A = CS.getFreshVar(Name + Stars, VarAtom::V_Other);
+    vars.push_back(A);
```

Comment:
This is speculating that you can find an instantiation that works. What if you can't?

Reply:
I guess the question is: If you can't, do you need to come back and undo some of this work later? Maybe the answer is 'no' because if you can't then the pointer will end up being WILD anyway?

Reply:
Yes. In `TypeVariableAnalysis` constraint variables are explicitly made WILD if the type variable cannot be instantiated.

Reply:
OK